_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,170 @@
+
+Search relevance was quantified using the industry-standard Normalized Discounted Cumulative Gain at rank 10 (NDCG@10). We also tracked system performance using search latency measurements. This setup provided a strong foundation for evaluating both search quality and operational efficiency.
+
+
+### NDCG@10
+
+|dataset	|Hybrid (min max)	|Hybrid (z score)	|Percent diff	|
+|---	|---	|---	|---	|
+|scidocs	|0.1591	|0.1633	|+2..45%	|
+|fiqa	|0.2747	|0.2768	|0.77%	|
+|nq	|0.3665	|0.374	|2.05%	|
+|arguana	|0.4507	|0.467	|	|
+|	|	|Average	|1.765	|
+
+### Search latency
+
+
+The following table presents search latency measurements in milliseconds at different percentiles (p50, p90, and p99) for both the Hybrid with min max and z score approaches. The *Percent difference* columns show the relative performance impact between these methods.
+
+<table> <tr> <th></th> <th colspan="3"><b>p50</b></th> <th colspan="3"><b>p90</b></th> <th colspan="3"><b>p99</b></th> </tr> <tr> <td></td> <td><b>Hybrid (min max)</b></td> <td><b>Hybrid (z score)</b></td> <td><b>Percent difference</b></td> <td><b>Hybrid (min max)</b></td> <td><b>Hybrid (z score)</b></td> <td><b>Percent difference</b></td> <td><b>Hybrid (min max)</b></td> <td><b>Hybrid (z score)</b></td> <td><b>Percent difference</b></td> </tr> <tr> <td>scidocs</td> <td>76.25</td> <td>77.5</td> <td>1.64%</td> <td>99</td> <td>100.5</td> <td>1.52%</td> <td>129.54</td> <td>133.04</td> <td>2.70%</td> </tr> <tr> <td>fiqa</td> <td>80</td> <td>81</td> <td>1.25%</td> <td>104.5</td> <td>105</td> <td>0.48%</td> <td>123.236</td> <td>124</td> <td>0.62%</td> </tr> <tr> <td>nq</td> <td>117</td> <td>117</td> <td>0%</td> <td>140</td> <td>140</td> <td>0%</td> <td>166.74</td> <td>165.24</td> <td>-0.90%</td> </tr> <tr> <td>arguana</td> <td>349</td> <td>349</td> <td>0%</td> <td>382</td> <td>382</td> <td>0%</td> <td>417.975</td> <td>418.475</td> <td>0.12%</td> </tr> <tr> <td></td> <td></td> <td><b>Average:</b></td> <td>0.72%</td> <td></td> <td><b>Average:</b></td> <td>0.50%</td> <td></td> <td><b>Average:</b></td> <td>0.64%</td> </tr> </table>
+
+### Conclusions
+
+
+Our benchmark experiments highlight the following advantages and trade-offs of Z-score normalization compared to min-max normalization in hybrid search approaches:
+
+**Search quality (measured using NDCG@10 across four datasets)**:
+
+* Z-score normalization shows a modest improvement in search quality, with an average increase of 1.765% in NDCG@10 scores.
+* This suggests that Z-score normalization may provide slightly better relevance in search results compared to min-max normalization.
```

Comment:
mention that min_max is out default, this will give a reason why we're comparing z-score with it